lib/api/route/streams.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:json_annotation/json_annotation.dart';
+
+import '../core.dart';
+part 'streams.g.dart';
+
+/// https://zulip.com/api/get-stream-topics
+Future<GetTopicsResult> getStreamTopics(
+  ApiConnection connection, {
+  required int streamId,
+}) {
+  return connection.get('getStreamTopics', GetTopicsResult.fromJson,
+    'users/me/$streamId/topics', {});
+}
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class GetTopicsResult {
+  final List<Topic> topics;
+
+  GetTopicsResult({
+    required this.topics,
+  });
+
+  factory GetTopicsResult.fromJson(Map<String, dynamic> json) =>
+    _$GetTopicsResultFromJson(json);
+
+  Map<String, dynamic> toJson() => _$GetTopicsResultToJson(this);
+}
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class Topic {
```

Comment:
Class names effectively live in one global namespace, not local to a file. (After all, this is a type we end up using in some other files.) So this name is much too general for this class, which has a fairly specialized purpose.

In particular, we have tons of variables and parameters etc. that mean a topic and are called "topic" — they all have type String, because that's how Zulip identifies topics.

One good name would be:

```suggestion
class GetStreamTopicsEntry {
```

Bit long, but it does the job of appropriately placing this class in context.